crates/hir_ty/src/lib.rs

```diff
@@ -1535,16 +1548,18 @@ mod tests {
     fn infer_function_param() {
         check_in_root_file(
             r#"
-                fn aaa(x: int, y: string) {
-                    let a = x
-                    let b = y
+                fn aaa(x: int, y: string, z: mut bool) {
+                    let a = x;
+                    let b = y;
+                    let c = z;
                 }
             "#,
             expect![[r#"
                 //- /main.nail
-                fn aaa(x: int, y: string) -> () {
+                fn aaa(x: int, y: string, z: mut bool) -> () {
                     let a = param:x; //: int
                     let b = param:y; //: string
+                    let c = param:z; //: bool
```

Comment:
**Optimization Issue**: Addition of a new mutable boolean parameter 'z' in function 'fn aaa'. This increases the function's complexity and potential side effects, which might affect performance if 'z' is frequently modified or accessed in a multi-threaded context. <br> **Fix**: Review the necessity of the mutable parameter 'z'. If mutability is not required for 'z' or if the function does not benefit from this parameter being mutable, consider changing it to immutable to reduce potential side effects and improve thread safety. <br> **Code Suggestion**: 
 ```
 -                fn aaa(x: int, y: string, z: mut bool) {
 +                fn aaa(x: int, y: string, z: bool) {
 ```